### PR TITLE
feat(employee): make a configurable birthday email template

### PR DESCRIPTION
**Pull request by DeeMysterio**
**previous system:**
It sent a birthday reminder email to all the employees except the one celebrating their birthday.

Screenshot 2019-11-27 at 7 41 12 PM

**The new system:**

- It sends out an email to all the employees including the one celebrating their birthday.

- It sends out an email per birthday guy. 

- Added a field in "HR Settings" saying "Birthday Email Template" to select an Email Template which if set, is used by the system to send out an email and has all the context about the employee otherwise a hard coded email message and subject is used.
 
hr

- The Email Template that is used to be used by existing(patches) and new sites(install_fixtures):

Screenshot 2019-11-22 at 5 40 26 PM

- A birthday email sent using a template designed with the available employee context in the Email Template:

Screenshot 2019-11-22 at 3 33 04 PM

### CODE REVIEW

File: erpnext/hr/doctype/employee/employee.py [hunk 1]
```diff
@@ -261,67 +261,48 @@ def update_user_permissions(doc, method):
...
-		birthdays.append({'company_email': '','employee_name': '','personal_email': '','user_id': ''})
-
-		for e in birthdays:
-			if e['company_email'] or e['personal_email'] or e['user_id']:
-				if len(birthday_names) == 1:
-					continue
-				recipients = e['company_email'] or e['personal_email'] or e['user_id']
+	employees = get_employees_born_today()
+	if employees:
+		recipients_list = frappe.get_all('Employee', filters={'status': 'Active'})
```
Comment (Anurag810) [inline erpnext/hr/doctype/employee/employee.py]: Get the recipients from the company where the birthday boys/girls belong. 
You can also handle multi-company Employee birthdays. where two employees having birthday belongs to two different companies, then they should have different recipients for different companies.

File: erpnext/hr/doctype/hr_settings/hr_settings.json [hunk 2]
```diff
@@ -160,13 +162,25 @@
...
+   "options": "Email Template"
   }
  ],
  "icon": "fa fa-cog",
  "idx": 1,
  "issingle": 1,
- "modified": "2019-08-05 13:07:17.993968",
- "modified_by": "Administrator",
+ "modified": "2019-11-27 17:41:39.717584",
+ "modified_by": "diksha@digithinkit.com",
```
Comment (Anurag810) [inline erpnext/hr/doctype/hr_settings/hr_settings.json]: Change modified_by to "Administrator"

**Review by Anurag810 — CHANGES_REQUESTED** [review]

**Comment by Anurag810** [comment]
Resolve Conflicts and update your Pull Request

**Comment by DeeMysterio** [comment]
@nabinhait This PR was closed after a discussion with @rmehta

**Comment by nabinhait** [comment]
@DeeMysterio okay, can you summarize the discussion?

I thought it got closed by bot automatically. 
And I was thinking that there might be separate emails for the birthday boy/girl and their colleagues.

**Comment by DeeMysterio** [comment]
> @DeeMysterio okay, can you summarize the discussion?
> 
> I thought it got closed by bot automatically.
> And I was thinking that there might be separate emails for the birthday boy/girl and their colleagues.

We can give different messages to the birthday guy/girl and their colleagues but they can anyway send personal wishes to the former if they want to. Anyway, the reason why this system got rejected is that I think he didn't like this design as it's sending automated birthday wishes and it can be annoying😶. Not sure as the discussion happened long back. You can confirm with him.

**Review by Anurag810 — CHANGES_REQUESTED** [review]
please resolve conflicts. and changes requested
